src/system-trace.js

```diff
@@ -0,0 +1,198 @@
+	};
+
+	var esImportDepsExp = /import [\s\S]*?["'](.+)["']/g;
+	var esExportDepsExp = /export .+ from ["'](.+)["']/g;
+	var commentRegEx = /(?:(?:^|\s)\/\/(.+?)$)|(?:\/\*([\S\s]*?)\*\/)/gm;
+	var stringRegEx = /(?:("|')[^\1\\\n\r]*(?:\\.[^\1\\\n\r]*)*\1|`[^`]*`)/g;
+
+	function getESDeps(source) {
+		var cleanSource = source.replace(commentRegEx, "");
+
+		esImportDepsExp.lastIndex = commentRegEx.lastIndex =
+			esExportDepsExp.lastIndex = stringRegEx.lastIndex = 0;
+
+		var match;
+		var deps = [];
+		var stringLocations = []; // track string for unminified source
+
+		function inLocation(locations, match) {
+		  for (var i = 0; i < locations.length; i++)
+			if (locations[i][0] < match.index && locations[i][1] > match.index)
+			  return true;
+		  return false;
+		}
+
+		function addDeps(exp) {
+			while (match = exp.exec(cleanSource)) {
+			  // ensure we're not within a string location
+			  if (!inLocation(stringLocations, match)) {
+				var dep = match[1];
+				deps.push(dep);
+			  }
+			}
+		}
+
+		if (source.length / source.split('\n').length < 200) {
+		  while (match = stringRegEx.exec(cleanSource))
+			stringLocations.push([match.index, match.index + match[0].length]);
+		}
+
+		addDeps(esImportDepsExp);
+		addDeps(esExportDepsExp);
+
+		return deps;
+	}
+
+	var instantiate = loader.instantiate;
+	loader.instantiate = function(load){
+		this._traceData.loads[load.name] = load;
+		var loader = this;
+		var instantiatePromise = Promise.resolve(instantiate.apply(this, arguments));
+
+		function finalizeResult(result){
+			var preventExecution = loader.preventModuleExecution &&
+				!isAllowedToExecute.call(loader, load);
+
+			// deps either comes from the instantiate result, or if an
+			// es6 module it was found in the transpile hook.
+			var deps = result ? result.deps : load.metadata.deps;
+
+			return Promise.all(map.call(deps, function(depName){
+				return loader.normalize(depName, load.name);
+			})).then(function(dependencies){
+				load.metadata.deps = deps;
+				load.metadata.dependencies = dependencies;
+
+				if(preventExecution) {
+					return {
+						deps: deps,
+						execute: emptyExecute
+					};
+				}
+
+				return result;
+
+			});
+		}
+
+		return instantiatePromise.then(function(result){
+			// This must be es6
+			if(!result) {
+				var deps = getESDeps(load.source);
+				load.metadata.deps = deps;
+			}
+			return finalizeResult(result);
+		});
+	};
+
+	var transpile = loader.transpile;
+	// Allow transpile to be memoized, but only once
+	loader.transpile = function(load){
+		var transpiled = load.metadata.transpiledSource;
+		if(transpiled) {
+			delete load.metadata.transpiledSource;
+			return Promise.resolve(transpiled);
+		}
+		return transpile.apply(this, arguments);
+	};
+
+	loader.eachModule = function(cb){
+		for (var moduleName in this._loader.modules) {
+			cb.call(this, moduleName, this.get(moduleName));
+		}
+	};
+}
+
+if(typeof System !== "undefined") {
+	applyTraceExtension(System);
```

Comment:
we have that new `addStealExtension` method that should be called if adding a new extension to steal
for example, see
https://github.com/stealjs/steal/blob/f8114d03f4ee4e085f87cd746fdc1c9130238726/src/cache-bust/cache-bust.js#L4`